packages/edit-site/src/components/page-pages/index.js

```diff
@@ -326,11 +332,64 @@ export default function PagePages() {
 				header: __( 'Date' ),
 				id: 'date',
 				render: ( { item } ) => {
-					const formattedDate = dateI18n(
-						getSettings().formats.datetimeAbbreviated,
-						getDate( item.date )
+					const isDraftOrPrivate = [ 'draft', 'private' ].some(
+						( status ) => status === item.status
 					);
-					return <time>{ formattedDate }</time>;
+					if ( isDraftOrPrivate ) {
+						return (
+							<>
+								{ __( 'Modified: ' ) }
+								<time>{ getFormattedDate( item.date ) }</time>
+							</>
+						);
+					}
+
+					const isPending = item.status === 'pending';
+					if ( isPending ) {
+						const isModified =
+							getDate( item.modified ) > getDate( item.date );
+						const dateToDisplay = isModified
+							? item.modified
+							: item.date;
+						return (
+							<>
+								{ __( 'Modified: ' ) }
+								<time>
+									{ getFormattedDate( dateToDisplay ) }
+								</time>
+							</>
+						);
+					}
+
+					const isScheduled = item.status === 'future';
+					if ( isScheduled ) {
+						return (
+							<>
+								{ __( 'Scheduled: ' ) }
+								<time>{ getFormattedDate( item.date ) }</time>
+							</>
+						);
+					}
+
+					const isPublished = item.status === 'publish';
+					if ( isPublished ) {
+						const isModified =
+							getDate( item.modified ) > getDate( item.modified );
+						const dateToDisplay = isModified
```

Comment:
Nit: I guess we could reuse the code to check if is modified. I don't think we care about the order of checks here, since we check against explicit post statuses, so the modified check could be grouped together..